### PR TITLE
validate-colors: add conformance for color values in :root

### PR DESCRIPTION
**Pull request by mduvall**
This looks for color usage that is not hex in the `:root` decl, as in:

``` css
:root {
  --Foo-background: rgba(0, 0, 0, 0);
}
```

and suggests to use hex values instead, as in:

``` css
:root {
  --Foo-background: rgba(#000, 0);
}
```

r? @michelle

### CODE REVIEW

**Comment by mduvall** [comment]
This passes `npm test`, anything else I should do?

**Comment by michelle** [comment]
oof. can we do `--foo-background` for variables?

File: index.js [hunk 1]
```diff
@@ -88,7 +88,7 @@ function getSimpleRules(rules) {
     if (rule.rules) {
       simpleRules = simpleRules.concat(getSimpleRules(rule.rules));
     }
-    if (rule.type == 'rule') {
+    if (rule.type === 'rule') {
```
Comment (michelle) [inline index.js:91]: :+1:

**Comment by michelle** [comment]
Is it easy to add this for all rules v. just in `:root`?

**Comment by mduvall** [comment]
@michelle: would all rules be trumped by recommending var usage instead?

**Comment by michelle** [comment]
Oh, right!

**Comment by michelle** [comment]
I know we're not doing this at the moment, but maybe add a few lines in README about the rule + rationale (easy grep-ability, etc).

:shipit: :rabbit:!

**Comment by michelle** [comment]
just added you as collab :)

File: test/fixtures/all-invalid-root-colors.css [hunk 2]
```diff
@@ -0,0 +1,4 @@
+/** @define InvalidRootColor; use strict */
+:root {
+  --InvalidRootColor-bad-color: rgba(0, 0, 0, 0);
```
Comment (michelle) [inline test/fixtures/all-invalid-root-colors.css]: can you add an `hsla` test, actually?

**Comment by mduvall** [comment]
@michelle: doneskies! (is :rabbit: a :ship: now?! O_o). Will merge in a jiffy.

**Comment by michelle** [comment]
:rabbit: is the new :ship:, like doneskies is apparently the new donezo :<

**Comment by mduvall** [comment]
ᕕ( ᐛ )ᕗ

**Comment by SlexAxton** [comment]
Fun!

**Pull request merged**